### ISSUE TITLE
S3 bucket serverSideEncryption with kmsMasterKeyId makes bucket resource not ready

### ISSUE THREAD

**Issue by yshu2017** [crossplane/provider-aws#540]
### What happened?

Attempted to create S3 bucket with serverSideEncryption settings with kmsMasterKeyId

claim.yaml
```
apiVersion: storage.ccs.test.com/v1alpha1
kind: S3Bucket
metadata:
  name: claimed-s3-test
  namespace: default
spec:
  parameters:
    bucketName: claimed-s3-test
    region: us-east-2
    version_enabled: Enabled
    serverSideEncryptionConfig:
      kmsMasterKeyId: "XXXXXXXfb-b87c-4sfsf1-8ea5-baXXXXcsdfs0e"
      sseAlgorithm: "aws:kms"
  compositionSelector:
    matchLabels:
      provider: aws
 ```

composition.yaml
```
apiVersion: apiextensions.crossplane.io/v1
kind: Composition
metadata:
  name: composites3buckets.aws.storage.ccs.test.com
  labels:
    guide: s3bucket
    provider: aws
    vpc: default
spec:
  writeConnectionSecretsToNamespace: crossplane-system
  compositeTypeRef:
    apiVersion: storage.ccs.test.com/v1alpha1
    kind: CompositeS3Bucket
  patchSets:
    - name: Metadata
      patches:
      - fromFieldPath: metadata.annotations
  resources:
    - base:
        apiVersion: s3.aws.crossplane.io/v1beta1
        kind: Bucket
        spec:
          forProvider:
            acl: private
            locationConstraint: us-west-2
            objectLockEnabledForBucket: false
            serverSideEncryptionConfiguration:
              rules:
                - applyServerSideEncryptionByDefault:
#                    kmsMasterKeyId: ""  #This key is ONLY applicable ONLY if sseAlgorithm is "aws:kms"
                    sseAlgorithm: AES256
      patches:
        - type: PatchSet
          patchSetName: Metadata
        - fromFieldPath: "spec.parameters.region"
          toFieldPath: "spec.forProvider.locationConstraint"
        - fromFieldPath: "spec.parameters.serverSideEncryptionConfig"
          toFieldPath: "spec.forProvider.serverSideEncryptionConfiguration.rules[0].applyServerSideEncryptionByDefault"
```

definition.yaml
```apiVersion: apiextensions.crossplane.io/v1
kind: CompositeResourceDefinition
metadata:
  name: composites3buckets.storage.ccs.test.com
spec:
  group: storage.ccs.test.com
  names:
    kind: CompositeS3Bucket
    plural: composites3buckets
  claimNames:
    kind: S3Bucket
    plural: s3buckets
  versions:
  - name: v1alpha1
    served: true
    referenceable: true
    schema:
      openAPIV3Schema:
        type: object
        properties:
          spec:
            type: object
            properties:
              parameters:
                type: object
                properties:
                  region:
                    type: string
                    description: "Specifies the Region where the bucket will be created. default to 'us-west-2'."
                  serverSideEncryptionConfig:
                    type: object
                    properties:
                      kmsMasterKeyId:
                        type: string
                        description: "(Optional) AWS Key Management Service (KMS) customer master key ID to use for the default encryption.
                        This parameter is allowed if and only if SSEAlgorithm is set to 'aws:kms'. Must be full arn like 'arn:aws:kms:us-east-2:123456789123:alias/aws/s3'
                        or keyId like 'abcdefg6-22x68-4a39-b07a-9b3dxxxx6c908'. Can't use 'alias/aws/s3'. No default set. "
                      sseAlgorithm:
                        type: string
                        description: "AWS Key Management Service (KMS) customer master key ID to use for the default encryption.
                        The ONLY allowed values are 'AES256' and 'aws:kms'. Default is 'aws:kms'"
                        enum: ["aws:kms", "AES256"]
                  
```
The bucket was created successfully. However, when describing s3Bucket and Bucket, their "READY" status are "false". 

### How can we reproduce it?

Try to create S3 bucket with above manifests

### What environment did it happen in?
Crossplane version: 

crossplane/provider-aws-controller:v0.16.0

**Comment by prasek** [comment]
@yshu2017 can you `kubectl describe` the s3bucket and Bucket so we can see what's going on?
/cc @hasheddan @krishchow

**Comment by yshu2017** [comment]
@prasek here is the "kubectl describe S3bucket..." result:

```
Name:         claimed-s3-test-yl
Namespace:    warrior
Labels:       <none>
Annotations:  crossplane.io/external-name: claimed-s3-test-yl
API Version:  storage.ccs.test.com/v1alpha1
Kind:         S3Bucket
Metadata:
  Creation Timestamp:  2021-02-19T19:20:35Z
  Finalizers:
    finalizer.apiextensions.crossplane.io
  Generation:        2
  Resource Version:  1578900
  Self Link:         /apis/storage.ccs.test.com/v1alpha1/namespaces/warrior/s3buckets/claimed-s3-test-yl
  UID:               aa184ac0-79a2-45e1-83cb-4b42fe30ec24
Spec:
  Composition Selector:
    Match Labels:
      Provider:  aws
  Parameters:
    Region:  us-east-2
    Server Side Encryption Config:
      Kms Master Key Id:  xxxxx1-xxx4f-xxx20-axxx3-xxxxxxcde04004
      Sse Algorithm:      aws:kms
  Resource Ref:
    API Version:  storage.ccs.test.com/v1alpha1
    Kind:         CompositeS3Bucket
    Name:         claimed-s3-test-yl-4n4c4
Status:
  Conditions:
    Last Transition Time:  2021-02-19T19:21:05Z
    Reason:                Composite resource claim is waiting for composite resource to become Ready
    Status:                False
    Type:                  Ready
Events:
  Type     Reason                      Age                 From                                                             Message
  ----     ------                      ----                ----                                                             -------
  Warning  BindCompositeResource       113s                offered/compositeresourcedefinition.apiextensions.crossplane.io  cannot update composite resource: Operation cannot be fulfilled on composites3buckets.storage.ccs.test.com "claimed-s3-test-yl-4n4c4": the object has been modified; please apply your changes to the latest version and try again
  Warning  ConfigureCompositeResource  113s                offered/compositeresourcedefinition.apiextensions.crossplane.io  cannot patch object: Operation cannot be fulfilled on composites3buckets.storage.ccs.test.com "claimed-s3-test-yl-4n4c4": the object has been modified; please apply your changes to the latest version and try again
  Normal   ConfigureCompositeResource  83s (x3 over 113s)  offered/compositeresourcedefinition.apiextensions.crossplane.io  Successfully applied composite resource
  Normal   BindCompositeResource       83s (x2 over 83s)   offered/compositeresourcedefinition.apiextensions.crossplane.io  Composite resource is not yet ready

```

**Comment by yshu2017** [comment]
sorry, missed the "describe bucket..."   Please check the following for it:

```
Name:         claimed-s3-test-yl-qsmlq-7nf5t
Namespace:    
Labels:       crossplane.io/claim-name=claimed-s3-test-yl
              crossplane.io/claim-namespace=warrior
              crossplane.io/composite=claimed-s3-test-yl-qsmlq
Annotations:  crossplane.io/external-name: claimed-s3-test-yl
API Version:  s3.aws.crossplane.io/v1beta1
Kind:         Bucket
Metadata:
  Creation Timestamp:  2021-02-22T16:36:46Z
  Finalizers:
    finalizer.managedresource.crossplane.io
  Generate Name:  claimed-s3-test-yl-qsmlq-
  Generation:     1
  Owner References:
    API Version:     storage.ccs.test.com/v1alpha1
    Controller:      true
    Kind:            CompositeS3Bucket
    Name:            claimed-s3-test-yl-qsmlq
    UID:             7b35a599-06f7-4cce-b1d2-db2b13172cdb
  Resource Version:  53275
  Self Link:         /apis/s3.aws.crossplane.io/v1beta1/buckets/claimed-s3-test-yl-qsmlq-7nf5t
  UID:               7c7c967d-a31c-455b-92ed-47167984ffca
Spec:
  For Provider:
    Acl:                             private
    Location Constraint:             us-east-2
    Object Lock Enabled For Bucket:  false
    Server Side Encryption Configuration:
      Rules:
        Apply Server Side Encryption By Default:
          Kms Master Key Id:  XXXXXXd12f1-XXXX-4XXXX-aXXX3-2114cXXXX0XXXX
          Sse Algorithm:      aws:kms
  Provider Config Ref:
    Name:  aws-provider
Status:
  At Provider:
    Arn:  arn:aws:s3:::claimed-s3-test-yl
  Conditions:
    Last Transition Time:  2021-02-22T16:36:46Z
    Reason:                Creating
    Status:                False
    Type:                  Ready
    Last Transition Time:  2021-02-22T16:36:47Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
Events:
  Type    Reason                   Age                From                                 Message
  ----    ------                   ----               ----                                 -------
  Normal  CreatedExternalResource  56s                managed/bucket.s3.aws.crossplane.io  Successfully requested creation of external resource
  Normal  UpdatedExternalResource  25s (x3 over 55s)  managed/bucket.s3.aws.crossplane.io  Successfully requested update of external resource

```

**Comment by jbhennin** [comment]
@muvaf I noticed that in terraform if you left the kmsMasterKeyId = " " that it would default to using the AWS managed keys for S3.

**Comment by muvaf** [comment]
@krishchow could you take a look at this?

**Comment by bandesz** [comment]
Ignore my previous comment if you saw it, I think this was fixed in https://github.com/crossplane/provider-aws/pull/497 and the fix was released in v0.17.0

**Comment by haarchri** [comment]
fixed with #497
